### PR TITLE
Add Mandrel to JDK family

### PR DESCRIPTION
**Pull request by AlexKovynev**
The only binary version on github for now

### CODE REVIEW

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package io.sdkman.changelogs.java
+
+import com.github.mongobee.changeset.{ChangeLog, ChangeSet}
+import com.mongodb.client.MongoDatabase
+import io.sdkman.changelogs.{
+  Mandrel,
+  Linux64,
+  MacOSX,
```
Comment (eddumelendez) [inline src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala:8]: it is not needed

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  Version,
+  Windows,
```
Comment (eddumelendez) [inline src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala:10]: it is not needed

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  removeVersion
```
Comment (eddumelendez) [inline src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala:11]: it is not needed

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        candidate = "java",
+        version = "20.1.0.1-mandrel",
+        url =
+          "https://github.com/graalvm/mandrel/releases/download/mandrel-20.1.0.1.Final/mandrel-java11-linux-amd64-20.1.0.1.Final.tar.gz",
+        platform = Linux64,
+        vendor = Some(Mandrel)
+      )
+    ).validate()
+      .insert()
+    setCandidateDefault("java", "20.1.0.1-mandrel")
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala:34]: I don't think we want to make the this the default for _all_ JDKs on SDKMAN :-)

**Comment by marc0der** [comment]
LGTM apart from setting it to be default.

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  @ChangeSet(
+    order = "001",
+    id = "001-add_mandrel_20_1_0_1",
+    author = "AlexKovynev"
+  )
+  def migrate001(implicit db: MongoDatabase) =
+    List(
+      Version(
+        candidate = "java",
+        version = "20.1.0.1-mandrel",
```
Comment (eddumelendez) [inline src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala:26]: Currently, the version just support 15 characters.

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala (hunk 4 above)
Comment (eddumelendez) [inline src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala:34]: not sure about doing so.

**Review by eddumelendez — CHANGES_REQUESTED** [review]
@AlexKovynev thanks for your contribution. I left some comments.

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala (hunk 5 above)
Comment (marc0der) [reply src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala:26]: I pushed it to 17 characters sometime ago for JDKs so this should be fine.

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala (hunk 2 above)
Comment (AlexKovynev) [reply src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala]: Fixed

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala (hunk 3 above)
Comment (AlexKovynev) [reply src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala]: Fixed

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala (hunk 1 above)
Comment (AlexKovynev) [reply src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala]: Fixed

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala (hunk 4 above)
Comment (AlexKovynev) [reply src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala]: Fixed

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala (hunk 5 above)
Comment (AlexKovynev) [reply src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala:23]: Yep, the same length in SAP Jdk now. Stay the same

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala (hunk 4 above)
Comment (AlexKovynev) [reply src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala]: Fixed

**Review by eddumelendez — APPROVED** [review]

**Review by marc0der — APPROVED** [review]

**Comment by AlexKovynev** [comment]
Aaaand someone can merge it?:)

File: src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import com.github.mongobee.changeset.{ChangeLog, ChangeSet}
+import com.mongodb.client.MongoDatabase
+import io.sdkman.changelogs.{
+  Mandrel,
+  Linux64,
+  Version
+}
+
+@ChangeLog(order = "035")
+class MandrelVmMigrations {
```
Comment (eddumelendez) [inline src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala:12]: just notice that this is not the filename. can you fixe it?

Comment (AlexKovynev) [reply src/main/scala/io/sdkman/changelogs/java/MandrelMigrations.scala]: Yep, fixed

**Comment by eddumelendez** [comment]
@AlexKovynev thanks for fixing the comments quickly. This PR is now merged in master.

**Comment by AlexKovynev** [comment]
It's look something wrong, sdk list java gen exception without details:(

**Comment by marc0der** [comment]
I've pulled the version, seems like we need to do some more work on the backend first.

**Comment by helpermethod** [comment]
Mandrel can now be installed via SDKMAN!. Thank you for your contribution @AlexKovynev!

image

**Pull request merged**